lib/m3u8-parser.js

```diff
@@ -30,17 +55,35 @@ var m3u8parser = module.exports = function() {
   this._write = function(chunk, encoding, callback) {
     var lines = chunk.toString('utf8').split('\n');
     if (lastLine) { lines[0] = lastLine + lines[0]; }
-    lines.forEach(function(line, i) {
-      if (i < lines.length - 1) {
-        parseLine(line);
-      } else {
-        // Save the last line in case it has been broken up.
-        lastLine = line;
+    if (this.isYtdlCore) {
+      for (var i = 0; i < lines.length; i++) {
+        var line = lines[i];
+        if (i < lines.length - 1) {
+          self.ytdlSavedLines.push(line);
+        } else {
+          // Save the last line in case it has been broken up.
+          lastLine = line;
+        }
       }
-    });
+    } else {
```

Comment:
Keep the old functionality just in case somebody wants to use this for something other than Ytdl-Core. m3u8 is very complicated.